CDCgov/prime-reportstream
# 6101 FHIR to HL7 library configuration reader

This PR adds the capability for the new FHIR to HL7 library to read and validate its configuration file per the doc (
[FHIR to HL7 Configuration Design.pdf](https://github.com/CDCgov/prime-reportstream/files/9123299/FHIR.to.HL7.Configuration.Design.pdf)
).  Consideration was taken that one day the config will reside in the database.

Notes on validation:
1. Field requirements are validated
2. FHIR path syntax is validated
3. HL7 Spec syntax CANNOT be validated as it is dependent on having data in a real message

Test Steps:
1. Run the build and tests by running the command `./gradlew package`
2. Verify the unit test pass.

## Changes
- Read and validate the library config file

## Checklist

### Testing

### Process

## Linked Issues
- #6101

## To Be Done

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
